Assets/HotmaxEditorVR/cloneControl.cs

```diff
@@ -4,38 +4,59 @@
 
 public class cloneControl : MonoBehaviour
 {
-    public objectSelect objSelect;
-    editStateController stateController;
-    public float scaleSize = .5f;
+    private bool cloneMode = false;
+
+    private bool _selectedObjectIsActive = stateManager.selectedObjectIsActive;
+    private GameObject _selectedObject;
+    private stateManager.editorModes _editorMode = stateManager.editorMode;
 
     private void Start()
     {
-        stateController = GetComponent<editStateController>();
-        objSelect.trackedController2.TriggerClicked += triggerClicked;
+        stateManager.editorModeEvent += updateEditorMode;
+        stateManager.selectedObjectIsActiveEvent += updateSelectedObjectIsActive;
+        stateManager.selectedObjectEvent += updateSelectedObjectEvent;
+
+        cloneSelectedObject();
+    }
+
+    protected virtual void OnApplicationQuit()
+    {
+        stateManager.editorModeEvent -= updateEditorMode;
+        stateManager.selectedObjectIsActiveEvent -= updateSelectedObjectIsActive;
     }
 
-    private void OnDestroy()
+    void updateEditorMode(stateManager.editorModes value)
     {
-        objSelect.trackedController2.TriggerClicked -= triggerClicked;
+        _editorMode = value;
     }
 
-    void triggerClicked(object sender, ClickedEventArgs e)
+    void updateSelectedObjectIsActive(bool value)
     {
-        if (objSelect.trackedController2.gripped)
+        _selectedObjectIsActive = value;
+        if (_selectedObjectIsActive)
+        {
+            cloneSelectedObject();
+        }
+    }
+
+    void updateSelectedObjectEvent(GameObject value)
+    {
+        _selectedObject = value;
+    }
+
+    //clone the currently selected object and delete the editing behaviours from it
+    void cloneSelectedObject()
+    {
+        if (_editorMode == stateManager.editorModes.cloneDeleteMode)
```

Comment:
You should do any checks like this outside of the function - so  on line 36 is a better place. Basically a function should run only when it actually needs to.

Reply:
got it.